python/cudf/cudf/core/frame.py

```diff
@@ -62,6 +62,17 @@ def _gather(self, gather_map):
         result._copy_categories(self)
         return result
 
+    def _hash(self, initial_hash_values=None):
+        return libcudfxx.hash.hash(self, initial_hash_values)
```

Comment:
Unsure if it makes sense for this to return a Column, CuPy array, or frame object. Leaving a comment to note.